DSharpPlus.Commands/Processors/TextCommands/TextExtensions.cs

```diff
@@ -0,0 +1,136 @@
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_Content")] private static extern void _messageContentSetter(DiscordMessage message, string content);
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_Channel")] private static extern void _messageChannelSetter(DiscordMessage message, DiscordChannel channel);
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_ChannelId")] private static extern void _messageChannelIdSetter(DiscordMessage message, ulong channelId);
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_guildId")] private static extern void _messageGuildIdSetter(DiscordMessage message, ulong? guildId);
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_Author")] private static extern void _messageAuthorSetter(DiscordMessage message, DiscordUser author);
+    [UnsafeAccessor(UnsafeAccessorKind.Field, Name = "mentionedUsers")] private static extern ref List<DiscordUser> _messageMentionedUsersSetter(DiscordMessage message);
+    [UnsafeAccessor(UnsafeAccessorKind.Field, Name = "mentionedRoles")] private static extern ref List<DiscordRole> _messageMentionedRolesSetter(DiscordMessage message);
+    [UnsafeAccessor(UnsafeAccessorKind.Field, Name = "mentionedChannels")] private static extern ref List<DiscordChannel> _messageMentionedChannelsSetter(DiscordMessage message);
+
+    /// <summary>
+    /// Creates a shallow copy of a Discord message.
+    /// </summary>
+    /// <param name="message">The message to copy.</param>
+    /// <returns>A shallow copy of the message.</returns>
+    public static DiscordMessage Copy(this DiscordMessage message) => _messageConstructor(message);
+
+    /// <summary>
+    /// Creates a fake message created event args object with the specified content.
+    /// </summary>
+    /// <param name="context">The command context to use for the message.</param>
+    /// <param name="message">The message to use as a base for the new message.</param>
+    /// <param name="content">The content of the new message.</param>
+    /// <returns>The created message created event args object.</returns>
+    public static MessageCreatedEventArgs CreateFakeMessageEventArgs(CommandContext context, DiscordMessage message, string content)
+    {
+        ArgumentNullException.ThrowIfNull(context, nameof(context));
+        ArgumentNullException.ThrowIfNull(message, nameof(message));
+        ArgumentException.ThrowIfNullOrWhiteSpace(content, nameof(content));
+
+        // Create a copy of the message to modify to prevent modifying the original message.
+        // By modifying the original message, we could potentially cause issues with the message cache.
+        DiscordMessage messageCopy = message.Copy();
+
+        // Modify the copied message to contain the contents that the user wants.
+        messageCopy.ModifyMessageProperties(content, context.Client, context.User, context.Channel, context.Guild);

```

Comment:
Is there any event in which the current client is not the same as it already is, or does the copy-constructor just not assign the client?

Reply:
The copy constructor copies the client from the other message. The client is used here to populate mentions from other users or channels